src/seedu/addressbook/storage/AddressBookDecoder.java

```diff
@@ -0,0 +1,90 @@
+package seedu.addressbook.storage;
+
+import static seedu.addressbook.parser.Parser.PERSON_DATA_ARGS_FORMAT;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.regex.Matcher;
+
+import seedu.addressbook.data.AddressBook;
+import seedu.addressbook.data.exception.IllegalValueException;
+import seedu.addressbook.data.person.Address;
+import seedu.addressbook.data.person.Email;
+import seedu.addressbook.data.person.Name;
+import seedu.addressbook.data.person.Person;
+import seedu.addressbook.data.person.Phone;
+import seedu.addressbook.data.person.UniquePersonList;
+import seedu.addressbook.data.tag.Tag;
+import seedu.addressbook.storage.StorageFile.StorageOperationException;
+
+/**
+ * Decodes the storage data file into an {@code AddressBook} object.
+ */
+public class AddressBookDecoder {
+
+    /**
+     * Decodes {@code encodedAddressBook} into an {@code AddressBook} containing the decoded persons.
+     *
+     * @throws IllegalValueException if any of the fields in any encoded person string is invalid.
+     * @throws StorageOperationException if the {@code encodedAddressBook} cannot be decoded.
+     */
+    public static AddressBook decodeAddressBook(List<String> encodedAddressBook)
+            throws IllegalValueException, StorageOperationException {
+        final List<Person> decodedPersons = new ArrayList<>();
+        for (String encodedPerson : encodedAddressBook) {
+            decodedPersons.add(decodePersonFromString(encodedPerson));
+        }
+        return new AddressBook(new UniquePersonList(decodedPersons));
+    }
+
+    /**
+     * Decodes {@code encodedPerson} into a {@code Person}.
+     *
+     * @throws IllegalValueException if any field in the {@code encodedPerson} is invalid.
+     * @throws StorageOperationException if {@code encodedPerson} cannot be decoded.
```

Comment:
Just a nit, the difference between the comments of these 2 exceptions aren't that clear. E.g. if `{@code encodedPerson} cannot be decoded`, does it mean it's `invalid`? So I think you can define the terms `invalid` vs `cannot be decoded` more clearly.